torch/_dynamo/variables/torch_function.py

```diff
@@ -138,10 +139,30 @@ class GetMethodMode(BaseTorchFunctionMode):
         """
 
         def __torch_function__(self, func, types, args=(), kwargs=None):
-            kwargs = kwargs or {}
-            nonlocal most_recent_func
-            most_recent_func = func
-            return func(*args, **kwargs)
+            ctx_manager = contextlib.ExitStack()
```

Comment:
TODO: Come back to this. Why is this running on cuda tensors at all, as opposed to cpu tensors?

Reply:
fixed it. See https://github.com/pytorch/pytorch/pull/140979#discussion_r1870619432